packages/cli/tests/commonlib/keyVaultValidator.ts

```diff
@@ -4,92 +4,195 @@
-  private static subscriptionId: string;
-  private static rg: string;
+  private projectPath: string;
+  private env: string;
 
-  public static init(ctx: any): IKeyVaultObject {
-    console.log("Start to init validator for Key Vault.");
+  private subscriptionId: string;
+  private rg: string;
+  private keyVault: IKeyVaultObject;
 
-    const resourceId = ctx[functionPluginName][functionAppResourceIdKeyName];
+  constructor(ctx: any, projectPath: string, env: string) {
+    console.log("Start to init validator for Key Vault.");
 
-    const keyVaultObject = {
-      m365ClientSecretReference: ctx[keyVaultPluginName][m365ClientSecretReferenceKeyName],
-      functionAppName: getSiteNameFromResourceId(resourceId),
-    } as IKeyVaultObject;
-    chai.assert.exists(keyVaultObject);
+    this.projectPath = projectPath;
+    this.env = env;
 
-    this.subscriptionId = ctx[solutionPluginName][subscriptionKey];
+    const resourceId = ctx[PluginId.KeyVault][StateConfigKey.keyVaultResourceId];
+    chai.assert.exists(resourceId);
+    this.subscriptionId = getSubscriptionIdFromResourceId(resourceId);
     chai.assert.exists(this.subscriptionId);
-
-    this.rg = ctx[solutionPluginName][rgKey];
+    this.rg = getResourceGroupNameFromResourceId(resourceId);
     chai.assert.exists(this.rg);
 
+    const keyVaultName = getKeyVaultNameFromResourceId(resourceId);
+    chai.assert.exists(keyVaultName);
+    this.keyVault = {
+      name: keyVaultName,
+      vaultUri: `https://${keyVaultName}.vault.azure.net`,
+    };
     console.log("Successfully init validator for Key Vault.");
-    return keyVaultObject;
   }
 
-  public static async validate(keyVaultObject: IKeyVaultObject) {
+  public async validate() {
     console.log("Start to validate Key Vault.");
 
     const tokenProvider = MockAzureAccountProvider;
     const tokenCredential = await tokenProvider.getAccountCredentialAsync();
     const token = (await tokenCredential?.getToken())?.accessToken;
 
-    console.log("Validating app settings.");
-
-    const response = await this.getWebappConfigReferenceAppSettings(
+    console.log("Validating key vault instance.");
+    const keyVaultResponse = await this.getKeyVault(
       this.subscriptionId,
       this.rg,
-      keyVaultObject.functionAppName,
+      this.keyVault.name,
       token as string
     );
+    chai.assert.exists(keyVaultResponse);
 
-    console.log("App Settings Key Vault References:");
-    console.log(response);
-
-    // Validate Key Vault reference in Azure Fucntion
-    chai.assert.exists(response);
-    chai.assert.equal(response.length, 1);
-    chai.assert.equal(response[0].name, "M365_CLIENT_SECRET");
-    chai.assert.equal(response[0].properties.secretName, "m365ClientSecret");
-    chai.assert.equal(response[0].properties.reference, keyVaultObject.m365ClientSecretReference);
-    chai.assert.equal(response[0].properties.status, "Resolved");
+    // Update permission
+    await this.updateKeyVaultPermission(
```

Comment:
minor: do we need to remove the permission after validation to avoid possible confusing (an unexpected account exists in ACL)?

Reply:
The entire resource group will be removed after the test case.

Reply:
Will consider remove the permission in next PR.